src/server/server_family.cc

```diff
@@ -2472,15 +2488,17 @@ void ServerFamily::Info(CmdArgList args, Transaction* tx, SinkReplyBuilder* buil
   }
 
   if (should_enter("REPLICATION")) {
-    util::fb2::LockGuard lk(replicaof_mu_);
     // Thread local var is_master is updated under mutex replicaof_mu_ together with replica_,
     // ensuring eventual consistency of is_master. When determining if the server is a replica and
     // accessing the replica_ object, we must lock replicaof_mu_. Using is_master alone is
     // insufficient in this scenario.
-    if (!replica_) {
-      append("role", "master");
-      append("connected_slaves", m.facade_stats.conn_stats.num_replicas);
+    util::fb2::LockGuard lk(replicaof_mu_);
+    bool is_master = bool(!replica_);
```

Comment:
why is the `bool` cast needed here? I'd expect the negative (`!`) to convert to boolean?

Reply:
probably not needed.